pkg/cpuallocator/allocator.go

```diff
@@ -343,6 +354,97 @@ func (a *allocatorHelper) allocate() cpuset.CPUSet {
 	return cpuset.NewCPUSet()
 }
 
+// getPackagePriorities calculates priority order between cpu packages for one
+// allocation request represented by allocatorHelper
+func (a *allocatorHelper) getPackagePriorities() map[sysfs.ID]int {
+	pkgs := a.sys.PackageIDs()
+	offline := a.sys.Offlined()
+
+	pkgAvail := make(map[sysfs.ID]int, len(pkgs))
+	pkgPrioAvail := make(map[sysfs.ID][NumCPUPriorities]int, len(pkgs))
+	for _, pkg := range pkgs {
+		avail := a.topology.pkg[pkg].Difference(offline).Intersection(a.from)
+		prioAvail := [NumCPUPriorities]int{}
+		for prio := PriorityHigh; prio < NumCPUPriorities; prio++ {
+			prioAvail[prio] = avail.Intersection(a.topology.cpuPriorities[prio]).Size()
```

Comment:
On e2e test VM (topology n4c16), I'm getting `a.topology.cpuPriorities[prio] == [empty, empty, empty]`. Therefore `iTotal` and `jTotal` are always zero. Even in the "stealing" case. As a result, this function ends up sorting packages based on the number of available cores (line 434).

Yet this still results in very nice CPU allocation for instance in the podpools e2e tests, I'd like to make sure that happens also in the case when not ending up (unintentionally?) to this fallback branch.